### PR TITLE
feat: add prover replay for easy debugging

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR creates a `prover_replay` program for easy debugging. When doing integrate tests, we run a Kroma node and assign tables again and again. For even a simple circuit like Tx circuit, it takes 5 minutes to create a proof. Another challenge we have involves debugging a c++ program from a rust binary, which can be done by attaching a debugger to a running process; however, this is tricky to implement. In conclusion, this PR will get rid of repeated table assignments and help us to debug a c++ program from a c++ binary by logging all the required data and re-running it from the data.

### CODE REVIEW

File: tachyon/c/zk/plonk/halo2/prover_impl_base.h [hunk 1]
```diff
@@ -42,7 +42,15 @@ class ProverImplBase : public tachyon::zk::plonk::halo2::Prover<PCS> {
   }
 
   void SetTranscript(
+      absl::Span<const uint8_t> state,
       std::unique_ptr<crypto::TranscriptWriter<Commitment>> writer) {
+    std::string_view state_str;
+    if (base::Environment::Get("TACHYON_TRANSCRIPT_STATE_LOG_PATH",
+                               &state_str)) {
+      VLOG(1) << "Save transcript state to: " << state_str;
+      CHECK(base::WriteFile(base::FilePath(state_str), state));
```
Comment (Insun35) [inline tachyon/c/zk/plonk/halo2/prover_impl_base.h:51]: Please include `tachyon/base/logging.h` for this.

File: tachyon/c/zk/plonk/halo2/prover_replay.cc [hunk 2]
```diff
@@ -0,0 +1,207 @@
+#include <string>
+
+#include "absl/strings/substitute.h"
+
+#include "tachyon/base/console/iostream.h"
+#include "tachyon/base/files/file_path_flag.h"
+#include "tachyon/base/files/file_util.h"
+#include "tachyon/base/flag/flag_parser.h"
```
Comment (Insun35) [inline tachyon/c/zk/plonk/halo2/prover_replay.cc:8]:
```suggestion
#include "tachyon/base/flag/flag_parser.h"
#include "tachyon/base/logging.h"
```

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**